In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import shapiro
import statsmodels.api as sm
from scipy.stats import mannwhitneyu

In [2]:
# Read case data
df = pd.read_excel("6. Case 5 - Mann-Whitney U test.xlsx")

In [3]:
# Print data head
df.head()

SalesPerson    New_Inc  Units_Sold
0            1  Incentive          27
1            2  Incentive          23
2            3  Incentive          23
3            4  Incentive          23
4            5  Incentive          25

In [4]:
# Create arrays with units sold for 'incentive' and 'no incentive'
data_incentive = df[df["New_Inc"] =="Incentive"]["Units_Sold"].values
data_no_incentive = df[df["New_Inc"] =="No Incentive"]["Units_Sold"].values

In [5]:
# Median units sold for 'incentive' and 'no incentive'
print("Median incentive: " + str(np.median(data_incentive)))
print("Median no incentive: " + str(np.median(data_no_incentive)))

Median incentive: 26.5
Median no incentive: 23.0


The median number of units sold is higher for sales reps who receive the new incentive (26.5) than for sales reps do no receive it (23.0). But is this difference of 3.5 units statistically significant? The Mann-Whitney can help us answer this question.

Since we are interested in the difference between the median units sold by the sales reps in the experimental and control groups, the null and alternative hypotheses can be formulated as follows:
- H0: Median (Units_Sold_Incentive) - Median (Units_Sold_NoIncentive) = 0 ("the median sales of the group of salespeople who did receive the incentive is equal to the median sales of the group of those who did not receive the incentive")

Furthermore, as the purpose of the test is to prove that the incentive program leads to a higher sales performance in terms of units sold, we can express the alternative hypothesis as:
- H1: Median (Units_Sold_Incentive) - Median (Units_Sold_NoIncentive) > 0 ("the median sales of the group of salespeople who did receive the incentive is greater than the median sales of the group of those who did not receive the incentive")

The **scipy.stats.mannwhitneyu()** function in SciPy performs the Mann-Whitney U test, a non-parametric statistical test designed to determine whether there is a significant difference between two independent samples. This test is particularly useful when the samples do not originate from populations that follow a normal distribution.

The main parameters of the **mannwhitneyu()** function include:

1. **a, b**: These arrays contain the sample data for comparison. Each array should contain the observations from one of the two independent samples.

2. **equal_var**: This parameter indicates whether the two input samples have equal variances. The default value is **True**, which assumes that the samples have equal variances. If set to **False**, the test will assume that the samples have unequal variances.

3. **alternative**: This parameter specifies the alternative hypothesis for the test. Options include:

    - **'two-sided'**: Tests if the two distributions are not equal (default). In this case, we're only interested in knowing if there's a difference between the two samples, regardless of the direction.
    - **'less'**: Tests if the distribution underlying a is stochastically less than the distribution underlying b. This means that F(u) > G(u) for all u, where F and G are the cumulative distribution functions of the samples a and b, respectively.
    - **'greater'**: Tests if the distribution underlying a is stochastically greater than the distribution underlying b. This means that F(u) < G(u) for all u, where F and G are the cumulative distribution functions of the samples a and b, respectively.
    
4. **method**: This parameter selects the method used to calculate the p-value. The available options are:

    - **'asymptotic'**: Compares the standardized test statistic against the normal distribution, correcting for ties. This method is suitable when the data has ties (i.e., identical values in the sample).
    - **'exact'**: Computes the exact p-value by comparing the observed U statistic against the exact distribution of the U statistic under the null hypothesis. No correction is made for ties. This method is appropriate for small sample sizes without ties.
    - **'auto'**: Automatically chooses the appropriate method based on the sample size and presence of ties. It selects 'exact' when one of the samples has fewer than eight observations and there are no ties; otherwise, it chooses 'asymptotic'.

Based on how the alternative hypothesis was formulated, we set **alternative='greater'**.

Then, since we are uncertain about which method to use, we specify the parameter **method='auto'**. This allows the function to automatically select the most appropriate method.

Furthermore, we set the significance level alpha to 0.05.

In [6]:
# Perform the Mann-Whitney U test
mannwhitneyu(data_incentive, data_no_incentive, alternative='greater', method='auto')

MannwhitneyuResult(statistic=145.0, pvalue=0.039888694711853884)

Since the p-value obtained from the Mann-Whitney U test (0.03989) is less than the alpha threshold (0.05), we can reject the null hypothesis in favor of the alternative. This establishes that the team members who received the new incentive sold more products (average of 26.5) compared to those who did not (average of 23.0). The difference of 3.5 units is statistically significant at the 0.05 level.